File: reverseString/reverseString.js
/**
 * Do not allocate extra space for another array, you must do this by modifying the input array in-place with O(1) extra memory.
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 *
 * Input: ["h","e","l","l","o"]
 * Output: ["o","l","l","e","h"]
 */
var reverseString = function(s) {
  let i = 0,
    j = s.length - 1;

  while (i < j) {
    // use ES6
    [s[i], s[j]] = [s[j], s[i]];
    i++;
    j--;
  }
};

File: reverseString/reverseString.py
class Solution(object):
    def reverseString(self, s):
        """
        :type s: List[str]
        :rtype: None Do not return anything, modify s in-place instead.

        用ES6 的语法 的 python解法
        """
        i = 0
        j = len(s) -1
        while i < j:
            [s[i], s[j]] = [s[j], s[i]]
            i = i + 1
            j = j -1
            